### PR TITLE
Integration with Swagger / OpenAPI specs

### PR DESCRIPTION
**Pull request by gunzip**
- Added API specs in YAML
- Added request and response validation with OpenAPI specs
- Added swagger routing

### CODE REVIEW

**Comment by cloudify** [comment]
I purposely left the ADRs in the `cittadinanza-digitale` repo, since they describe our overall project choices, whereas this repo only contains the code for Azure Functions.

File: tsconfig.json [hunk 1]
```diff
@@ -5,8 +5,8 @@
     "alwaysStrict": true,
     "noImplicitAny": true,
     "strictNullChecks": true,
-    "noUnusedLocals": true,
-    "noUnusedParameters": true,
+    "noUnusedLocals": false,
```
Comment (cloudify) [inline tsconfig.json]: we must keep the code tight, these settings must be kept enabled

File: lib/public_api_v1.ts [hunk 2]
```diff
@@ -2,50 +2,127 @@
  * Main entrypoint for the public APIs handlers
  */
 
-import * as express from "express";
+import * as express from 'express'
```
Comment (cloudify) [inline lib/public_api_v1.ts]: reason?

File: lib/public_api_v1.ts [hunk 3]
```diff
@@ -2,50 +2,127 @@
...
-import { CreateMessage, GetMessage, GetMessages } from "./controllers/messages";
-import { GetProfile, UpdateProfile } from "./controllers/profiles";
+import debugHandler from './controllers/debug'
+import { CreateMessage, GetMessage, GetMessages } from './controllers/messages'
+import { GetProfile, UpdateProfile } from './controllers/profiles'
 
 // Setup Express
 
-const app = express();
+const jsonRefs = require('json-refs')
```
Comment (cloudify) [inline lib/public_api_v1.ts]: `import * as jsonRefs from "json-refs"`

File: lib/public_api_v1.ts [hunk 4]
```diff
@@ -2,50 +2,127 @@
...
+app.post('/xapi/v1/messages/:fiscalcode', CreateMessage(messageModel))
+
+/////////////////
+
+const resolveSwaggerSpecs = (apiPath: string) =>
+  jsonRefs.resolveRefsAt(apiPath, {
+    // Resolve all remote references
+    filter: ['relative', 'remote'],
+    loaderOptions: {
+      processContent: (res: any, cb: any) =>
```
Comment (cloudify) [inline lib/public_api_v1.ts]: avoid using any in production code

File: lib/public_api_v1.ts [hunk 5]
```diff
@@ -2,50 +2,127 @@
...
+      res.status(500).json(validationResponse)
+    }
+  )
+
+const registerSwaggerMiddleware = (config: any) =>
+  new Promise((resolve, reject) => {
+    swaggerExpress.create(config, (err: Error, swaggerExpress: any) => {
+      if (err) {
+        reject(err)
+        return
```
Comment (cloudify) [inline lib/public_api_v1.ts]: use if/else instead of return

**Comment by gunzip** [comment]
refactoring everything

**Pull request closed without merging**